src/views/view.c

```diff
@@ -890,6 +890,103 @@ const GList *dt_view_get_images_to_act_on(const gboolean only_visible, const gbo
   return darktable.view_manager->act_on.images;
 }
 
+// get the query to retrieve images to act on. this is useful to speedup actions if they already use sqlite queries
+gchar *dt_view_get_images_to_act_on_query(const gboolean only_visible)
```

Comment:
if dt_selection_get_list_query() above is well named (returns the query), it seems the of name dt_view_get_images_to_act_on_query() is a bit misleading (returns the string list).

Reply:
More a small comment: column is with 1 l (in the comments)

Reply:
no, `dt_view_get_images_to_act_on_query` return either an sqlite subquery `SELECT ....` in case we return the selection either the list of id `1234,1235,...`
in both case this can be use as a subquery. Of course this only improve speed in case of selection, but in other cases the list is so small that it doesn't matter :)

Reply:
oops, I'd not seen it could return also a full query. Your are right.
I found it still a bit misleading... :)